transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/editor/internal/l10n/messages.properties

```diff
@@ -186,11 +186,13 @@ TransformationEditor_helpTextTarget=Create a new mapping below by dragging a var
 TransformationEditor_invalidPomConfiguration=Unable to recognize project's POM configuration.
 TransformationEditor_invalidTransformationFile=Invalid transformation file: Must be a Dozer configuration file.
 TransformationEditor_messageDialogConfirmation=Are you sure?\n\nAll incomplete mappings will be lost when the editor is closed.
+TransformationEditor_migrationDialogConfirmation=The transformation file must be converted to support the current Camel version.\n\nAre you sure you want to continue?
 TransformationEditor_mustBeOpenedViaCamelEditor=The transformation editor must be opened via the Camel editor.
 TransformationEditor_tooltipHideSourceVariableViewers=Hide the source/variables viewers
 TransformationEditor_tooltipHideTargetViewers=Hide the target viewer
 TransformationEditor_tooltipShowSourceVariableViewers=Show the source/variables viewers
 TransformationEditor_tooltipShowTargetViewers=Show the target viewer
+TransformationEditor_unableToOpenIncompatibleTransformationFile=Unable to open incompatible transformation file
```

Comment:
Why is it incompatible? It would be nice to precise to user.

Reply:
I think you're wanting to give too much detail to the user.  This file is supposed to be entirely handled under the covers by the tooling, and should NEVER be modified by the user.  I don't think the user should be concerned about the details of the incompatibility, but I could update the message to say it's related to not supporting Camel 2.2.0? WDYT?

Reply:
> This file is supposed to be entirely handled under the covers by the tooling, and should NEVER be modified by the user.

I think that it is possible for a user to modify this file as it is a normal Dozer file.
It might be even almost normal for Camel users which were not using the Fuse Tooling before. (I still hope that we are converting users :-) )

> I could update the message to say it's related to not supporting Camel 2.2.0? WDYT?

yes, would be nice to precise that the file need to be compatible with Dozer version used by Camel 2.20+

Reply:
it is definitely NOT the case that the Dozer file should ever be modified by the user. Despite being a Dozer file, it has always been the goal to hide this file from the user, especially since we add our own custom annotations/modifications to the file that Dozer does not inherently understand. Unless you feel strongly otherwise, I'll update the messages as you suggest without diving into particulars about the Dozer file contents.

Reply:
> we add our own custom annotations/modifications to the file that Dozer does not inherently understand.

it is a Camel Dozer stuff which are added? not Fuse Tooling specific?

Reply:
> it is definitely NOT the case that the Dozer file should ever be modified by the user

How Camel Dozer users are doing when not using Fuse Tooling? My point was to be able to welcome them and that this point will not block them to adopt Fuse Tooling.

Reply:
The changes we add the to file are indeed specific to the tooling. We've modified the runtime to be able to process these changes, but there is no facility provided to add/deal with these changes outside of the tooling. It's been like this from day one.

Reply:
The goal with Dozer has always been to provide a transformation engine that could, if we chose, be replaced under the covers someday with another engine with minimal impact to the user. Not sure if we really accomplished that, but that was the goal.